### ISSUE TITLE
Transforms sometimes fail resulting in a uncaught exception instead of an error

### ISSUE THREAD

**Issue by stkhapugin** [jsonmodel/jsonmodel#333]
`NSDate` can be converted to and from `NSString` by means of `__NSDateFromNSString` and `__JSONObjectFromNSDate`. However, if the JSON format changes to include something instead of a string (not necessarily a front-end developer's fault), say, an `NSDictionary`, instead of correctly issuing an error and returning nil from `initWithDictionary:error:`, an uncaught exception is thrown.  

Code to reproduce:
```objective-c
@interface User : JSONModel
@property (nonatomic, strong) NSDate *birthday;
@end
@implementation User @end

NSDictionary *goodJSON = @{@"birthday":@"1992-03-15 00:00:00.000000"};
NSError *error = nil;
User *test = [[User alloc] initWithDictionary:goodJSON error:&error];
// test is not nil, error is nil, everything is nice and dandy

NSDictionary *badJSON = @{@"birthday":@{@"date":@"1992-03-15 00:00:00.000000", @"time":@123}};
User *anotherTest = [[User alloc] initWithDictionary:badJSON error:&error]; //< throws an exception
```

Here's what the comment above the `@throw` says: "it's not a JSON data type, and there's no transformer for it if property type is not supported - that's a programmer mistaked -> exception". I believe that a format change on the backend may not always be a programmer's mistake and in this case an error should be issued instead of an exception. What'd you say?

---
actually here is what our backend sent me today: 
```JSON
"birthday":{
                    "date":"1992-03-15 00:00:00.000000",
                    "timezone":"Asia/Beijing",
                    "timezone_type":3
}
```

**Comment by billinghamj** [comment]
I agree that this should be handled rather than thrown, so we will look into trying to resolve this.

However, to actually handle the two inputs, you can also have two custom setters:

```objc
- (void)setBirthdayWithNSString:(NSString *)string;
- (void)setBirthdayWithNSDictionary:(NSDictionary *)dictionary;
```

**Comment by billinghamj** [comment]
Whoops sorry, need to keep it open as there is still a bug.

**Comment by huiyumiao** [comment]
Hi,I use this method 
`-(instancetype)initWithString:(NSString*)string error:(JSONModelError**)err;`
and I generate a json string :
`NSData *jsonData = [NSJSONSerialization dataWithJSONObject:dic options:NSJSONWritingPrettyPrinted error:&error];
NSString *jsonString = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];`
but it throws "it's not a JSON data type, and there's no transformer for it if property type is not supported ...". How can I fix it?

**Comment by billinghamj** [comment]
@huiyumiao Please open a new issue and provide the JSON string you're providing, the model/class code, and how you're invoking it. Do not use this issue to discuss your problem - it is not related to the issue raised here.

**Comment by billinghamj** [comment]
This is resolved in v1.5.1. Apologies that it took me so long to sort this out.
